### DISCUSSION TITLE
Welcome to getting-started Discussions!

### DISCUSSION THREAD

**Discussion by timea-solid** [solid-contrib/getting-started#3] [Announcements]
## 👋 Welcome!

**Comment by YetAnotherJonWilson** [comment]
Hello to everyone joining us to create the Solid Onboarding Experience for Developers!

Many thanks to Timea for setting up the repo, and getting the Discussions tool working.

I want to also welcome **everyone** who has an interest in this project, in developing Solid Apps, or in the Solid ecosystem more broadly. I hope that together, we can foster **inclusiveness**, **respect**, and **helpful, positive motivation**.

I'm Jon, I'm a maintainer of the project. My primary goal as a maintainer of this project is to enable and motivate developers and potential developers as they learn and build Solid Apps. I especially want to focus on creating a straightforward, logical path for someone to create their **first Solid app**, and to have clarity about how to learn and build more.

As a way of implementing that goal, I hope to build momentum in this project by: 
- concentrating efforts related to developer onboarding here, in this repo, so that everyone involved in this project can see all the efforts going on, and can join them or give feedback
  - for example, I've been working on creating a diagram using a [Miro board](https://miro.com/app/board/uXjVP6z2ZlM=/?share_link_id=191602903085)--while I can't work on  that board inside of this Github repo, I will try to create issues, links, discussions, and/or post about that effort in the wiki
  - as another example, I've been creating a large list of potential Solid apps in my own note-taking app (Obsidian), planning to _eventually_ include it on our web page, but in keeping with the prinicple of concentrating everything here, I'll create either a discussion topic, a wiki page, or an issue, and start adding those ideas so that people can start adding their own and discussing them (or even building them?!) now, rather than at some unspecified future date

**Comment by timea-solid** [comment]
Hei @YetAnotherJonWilson what is the scope of the Miro board diagram? what do you want to capture?

Related to the list of Solid apps - is this beyond the list on the website: https://solidproject.org/apps or the tutorial list: https://timea.solidcommunity.net/HelloWorld/ Since there are such efforts elsewhere, what would be the goal of your list?

**Comment by YetAnotherJonWilson** [comment]
Hi Timea,

### Regarding the scope of the Miro diagram, and what I want to capture

At first, my goal was to create a high-level overview of all the parts of Solid--WebId's, ID providers, Pods, Pod providers, Users, etc.--and show the interactions between the parts. The motivation for this was to create a **visual** aid to users who are more visual thinkers (like myself), to help clarify concepts in other parts of the documentation. 

Then, after creating my first iteration of a diagram, I came across a different [diagram](https://forum.solidproject.org/t/solid-ecosystem-diagram/6921)--posted on the forum by a user named Brent--that included many of the same parts, but had a very different style. This made me start to think about how to determine what is most important to show in the diagram, and how to best show it (in a way that makes sense, but is also approachable for newcomers and people like me who don't understand formal diagramming schemes like UML).

At the moment, I've paused updating that diagram because I've decided that the best use of my effort right now is to focus on creating an "MVP" version of this documentation page.

### Regarding the list of Solid Apps

For disambiguation, there are 2 things that "list of Solid Apps" might be referring to.

1) Solid App Ideas. 
In my note above, in this discussion, where I say, "as another example, I've been creating a large list of potential Solid apps", I am referring to a list of Solid apps that are just ideas: 1) the apps don't actually exist yet, and 2) the specific list I'm referring to exists only in my personal Obsidian notebook, where I have a list of about 150 ideas for different Solid apps, but 3) I think it would be very useful to share with new Solid developers **as a means of motivation, to show what might be possible with Solid, and to provide specific examples for developers who might want to create something, but don't know where to start.**

2) Existing Solid Apps. 
There are existing lists of Solid Apps--there are the two that you linked above (and _maybe_ others?). I suppose my intention here, on this page, would be **to make sure that the apps are findable by new developers** . . . but I guess this raises a couple questions: 1) are they in the scope of this project (which, perhaps, raises a larger question about the scope itself, and whether we should more formally define it); and 2) if they are in the scope of this project, and they should be included, then _how_ should we include them: a) individually, as examples, under one of  the diataxis categories (like, tutorials, maybe?), b) as a separate list of links (which is what I was leaning towards doing, but maybe I should rethink this), or c) by just providing links, including the two you posted above, and others (if there are others), so that developers can find them on their own, if they want to?

**Comment by danielbakas** [comment]
Hi everyone!! This is great! @YetAnotherJonWilson, for the past year we have been doing something similar (internally) to your Obsidian project at our Notion, to map Solid Apps, community members WebIDs and social networks, resources and much more.

Should we combine our knowledge into a single source? 🤩

Also, is this project being represented in RDF and Solid itself? I believe that it would be much more meaningful to represent all of this knowledge in Solid and RDF. Not only for us to understand the technology better, but also to showcase its potential 🤓

We'd love to know what you think 💡

**Comment by YetAnotherJonWilson** [comment]
Hey @danielbakas , 

As to your second question, well, if I understand what you're asking, the short answer is yes, but it's a work-in-progress. I am creating a list of documentation and related resources, stored in a [Solid Pod](https://onboarding.solidcommunity.net/public/Links), which will include existing apps. And I am tentatively planning to create a container in the same pod to keep track of App Ideas. I agree with you that understanding and showcasing the tech are good reasons to use Solid for this purpose.

I'm not sure I understand the scope of the project you're describing at Notion, and whether that's related what I'm trying to do--my goal is to create: 1) a (comprehensive, hopefully) list of resources to help developers get started, and 2) a list of existing Solid Apps, for the purpose of showing developers examples of what is being done, how it's being done, and if applicable, how they can help with any apps that could use help. I'm not sure how or why we would want to map WebID's or social networks (are there any that exist in the Solid ecosystem already?). WebID's, specifically, sound like something I wouldn't want to keep track of (though, to the extent they are publicly viewable, users should be aware that in theory, they could be collected). 

I don't think I've heard of Notion before (but maybe you mentioned it in one of the chats, and I forgot). Was the GPT bot that you built a part of that project, or is it unrelated?

Thanks for your comment and questions!

**Comment by danielbakas** [comment]
Hi @YetAnotherJonWilson and @timea-solid!

Thank you for your insights and questions! Let me clarify a bit about my Notion work:

1. **Project Scope**: For the past years I've been gathering a diverse array of resources related to the semantic web, linked data, and more recently, Solid. This includes projects, applications, high-level information about individuals (without ever compromising privacy), topics, organizations, various resources, and more. The goal is to map these elements, understand their interconnections, and get a high-level view of the ecosystem to provide value to our community.

2. **Privacy and WebIDs**: I completely agree with your privacy concerns regarding WebIDs. My approach involves reaching out to individuals who have already shared their WebIDs publicly, inviting them to be part of our "community map", as well of our spaces for sharing and showcasing. Additionally, we could extend this invitation to the broader Solid community, allowing anyone interested to share their WebIDs and projects.

3. **Integration with Your Work**: I'm eager to share my work from Notion, especially the structure and content. I believe collaborating, along with community input, could greatly enhance the design of onboarding resources and the showcase of existing Solid apps and developments.

4. **GPT Bot Project**: The GPT bot I developed is separate from the Notion project. It's trained on personal conversations and resources, but operates independently. However, all data from Notion is transferrable and could be integrated into our collaborative effort. Moreover, the GPT bot could become an asset for the Solid community, offering a platform for contribution and usage by everyone.

I'm genuinely excited about contributing further to this project and look forward to hearing both your and Timea's thoughts on these ideas.

Thank you for this space!

**Reply by YetAnotherJonWilson** [reply]
Thanks Daniel, that all sounds great! I am trying to get things to a point where others can easily come along and contribute without needing to coordinate with me at all. 

I think the first step in this regard will be for me to finish working on a form for submitting resources. As soon as that's done (fingers crossed I'll finish it this week...), you could use that to add any resources that you've discovered that are related to Solid. Sounds like you have a lot, so let me know if I can help out in that regard also.

Basically, the form will let you add, for each resource: 1) a URL, 2) a name (like "Building your first Solid App", or "Solid Authentication Protocol"), 3) whether it's a reference (aka protocol or spec) type document, a Tutorial, a How-to guide, or an Explanation of how something works, 4) whether the resource is still valid (optional), or 5) has been updated recently...and maybe a few other things, like who submitted the resource, and the datetime it was submitted, and when it was last checked for being up-to-date.

That will allow anyone to be able to help add to our collection of resources, which you can see here, by the way (I think I shared this with you already, but just in case): https://solid-contrib.github.io/getting-started/

And then I also want to start adding "Good first issues" here in this repo itself.

Oh, also, by the way, I used your GPT Bot again, and it was **so** helpful. Thanks again for that!

**Reply by danielbakas** [reply]
Hi @YetAnotherJonWilson!

That sounds phenomenal! I'd love to contribute with the form if you could use some help 🤓

Should we move this "Resources" discussion to another thread maybe?

So happy the GPT is being helpful!! Thank you for trying it out! Please let me know if there's anything you'd like for us to tweak or play around with, to make it even better 🤩

**Comment by LabObjects** [comment]
Would a Glossary of Terms be a good fit for this repo? I was reading through the Solid Forum and observed a request for clarification of terms/acronyms as some readers (such as myself) weren't familiar terms being used. I'd be happy to start it, but wanted to get feedback from this repo's contributors if they thought this would be the right place and if it would add value. Thanks.

**Reply by YetAnotherJonWilson** [reply]
Hi @LabObjects 

I'm sorry to take so long to respond--I will try to be much better about responding quickly going forward. Actually, that sounds like a great idea. If you'd like to get the ball rolling--by collecting some words, acronyms and terms, and beginning defining them--that would be terrific. You could do this in a plain text file, or, if you're familiar with JSON, that would be a format that would be simple to parse and use on our site. We could probably add a special section just for the glossary.

Please let me know if you'd like further feedback or anything. I'll create an issue, and we can continue the discussion there, as needed.

Thank you!

**Reply by YetAnotherJonWilson** [reply]
Here's the issue I created: https://github.com/solid-contrib/getting-started/issues/26

**Reply by LabObjects** [reply]
Hi Jon, Thanks! I will start this after Monday next week as things have suddenly piled up on me. My name is April Daly - I signed up for most of the W3C stuff with my company name. Thanks for your help and have a great weekend! April

**Reply by YetAnotherJonWilson** [reply]
Sounds great! Thanks for getting this going, I think it'll be very helpful. Have a wonderful weekend, also.

**Comment by bourgeoa** [comment]
There is already a small glossary here https://github.com/solid/solidcommunity.net/wiki/Solid-Environment-&-Glossary#glossary

It is open to updates.

**Reply by LabObjects** [reply]
Thank you, Alain!

**Comment by bourgeoa** [comment]
This could also be of interest https://jeff-zucker.solidcommunity.net/sp4/

**Reply by LabObjects** [reply]
Thank you, Alain!

**Comment by YetAnotherJonWilson** [comment]
@bourgeoa @LabObjects

This actually highlights one of the main problems with Solid that this repo was initially created to address--there are lots of resources out there, but it is **very** difficult for a beginner to find them. 

@LabObjects had a great idea--we should definitely have a glossary that someone can find on our page. 

@bourgeoa Are you suggesting that, rather than create our own glossary and host it here, we provide links to these existing glossaries? 

@timea-solid Do you have any input?

**Reply by LabObjects** [reply]
Hi, 

I was thinking a glossary for this repo would incorporate entries from the provided references. I agree with Jon's statement regarding the challenge to find information. The scope of the (new) glossary could also include topics that are related to Solid (i.e., lots of terms are used in the Solid forum where it is often assumed the reader knows them (e.g., https://forum.solidproject.org/t/vocabulary-helpdesk-resources/6714/5)). 

Thanks, April Daly

**Reply by YetAnotherJonWilson** [reply]
I agree, I think this is the direction to go.
